src/raw.rs

```diff
@@ -357,103 +396,97 @@ pub(crate) fn open_key_with_flags(
     mode: KeyMode,
     flags: c_int,
 ) -> *mut RedisModuleKey {
-    unsafe {
-        RedisModule_OpenKey.unwrap()(ctx, keyname, mode.bits() | flags).cast::<RedisModuleKey>()
-    }
+    redis_call!(RedisModule_OpenKey(ctx, keyname, mode.bits() | flags))
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 #[inline]
 pub fn reply_with_array(ctx: *mut RedisModuleCtx, len: c_long) -> Status {
-    unsafe { RedisModule_ReplyWithArray.unwrap()(ctx, len).into() }
+    redis_call!(RedisModule_ReplyWithArray(ctx, len))
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 #[inline]
 pub fn reply_with_map(ctx: *mut RedisModuleCtx, len: c_long) -> Status {
-    unsafe {
-        RedisModule_ReplyWithMap
-            .map_or_else(
-                || RedisModule_ReplyWithArray.unwrap()(ctx, len * 2),
-                |f| f(ctx, len),
-            )
-            .into()
-    }
+    unsafe { RedisModule_ReplyWithMap }
+        .map_or_else(
+            || redis_call!(RedisModule_ReplyWithArray(ctx, len * 2)),
+            |f| unsafe { f(ctx, len) },
+        )
+        .try_into()
+        .unwrap()
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 #[inline]
 pub fn reply_with_set(ctx: *mut RedisModuleCtx, len: c_long) -> Status {
-    unsafe {
-        RedisModule_ReplyWithSet
-            .map_or_else(
-                || RedisModule_ReplyWithArray.unwrap()(ctx, len * 2),
-                |f| f(ctx, len),
-            )
-            .into()
-    }
+    unsafe { RedisModule_ReplyWithSet }
+        .map_or_else(
+            || redis_call!(RedisModule_ReplyWithArray(ctx, len * 2)),
+            |f| unsafe { f(ctx, len) },
+        )
+        .try_into()
+        .unwrap()
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
 #[inline]
 pub fn reply_with_attribute(ctx: *mut RedisModuleCtx, len: c_long) -> Status {
-    unsafe { RedisModule_ReplyWithAttribute.unwrap()(ctx, len).into() }
+    redis_call!(RedisModule_ReplyWithAttribute(ctx, len))
 }
 
 #[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub fn reply_with_error(ctx: *mut RedisModuleCtx, err: *const c_char) {
-    unsafe {
-        let msg = Context::str_as_legal_resp_string(CStr::from_ptr(err).to_str().unwrap());
-        RedisModule_ReplyWithError.unwrap()(ctx, msg.as_ptr());
-    }
+pub fn reply_with_error<S: AsRef<str>>(ctx: *mut RedisModuleCtx, err: S) -> Status {
```

Comment:
This was used to get `*const c_char`, I believe it will not accept it now right? So its a breaking change. Another reason why this should be on the next major release.

Reply:
Yes, `*const char`; indeed, it breaks the compatibility. This is also why all of this implementation detail should be hidden behind `pub(crate)` too.

Reply:
Now, it is hidden behind `pub(crate)` (the whole module).